pandas/tools/merge.py

```diff
@@ -1411,6 +1411,10 @@ def __init__(self, objs, axis=0, join='outer', join_axes=None,
             sample = objs[0]
         self.objs = objs
 
+        # Check for string axis parameter
+        if isinstance(axis, str):
+            axis = objs[0]._get_axis_number(axis)
```

Comment:
iirc it's a class method so something like

NDFrame._get_axis_number(axis)  should work